### PR TITLE
Normative: Added support for sentence break suppressions to Intl.Segmenter

### LINKED ISSUE #580: Add support for more subtags
Currently, a number of Unicode extension keywords are supported, including `nu` for numbering systems, `ca` for calendaring systems, `hc` for hour cycles, etc. 

This issue tracks all the BCP-47 subtags added by Unicode that we do not yet support. Feel free to add to the list below this:

1. `rg` (Region override): see #370.
2. `fw` (First day of the week): see #6.

### PR DESCRIPTION
**Pull request by ben-allen**
fix #580 

Added support for u-ss to Intl.Segmenter

### CODE REVIEW

File: spec/segmenter.html [hunk 1]
```diff
@@ -74,11 +77,8 @@ <h1>Internal slots</h1>
       </p>
 
       <p>
-        The value of the [[RelevantExtensionKeys]] internal slot is &laquo; &raquo;.
+        The value of the [[RelevantExtensionKeys]] internal slot is &laquo; *"ss"* &raquo;.
       </p>
-      <emu-note>
-        Intl.Segmenter does not have any relevant extension keys.
-      </emu-note>
```
Comment (srl295) [inline spec/segmenter.html:81]: There are actually others: lb, lw, dx. See https://www.unicode.org/reports/tr35/#UnicodeLineBreakStyleIdentifier

**Review by srl295 — APPROVED** [review]
LGTM

File: spec/segmenter.html (hunk 1 above)
Comment (sffc) [reply spec/segmenter.html:81]: lb and lw are only for line break, which Intl.Segmenter doesn't support.

dx is not widely used or implemented, and I've raised questions about its utility.

File: spec/segmenter.html [hunk 2]
```diff
@@ -17,16 +17,19 @@ <h1>Intl.Segmenter ( [ _locales_ [ , _options_ ] ] )</h1>
...
         1. Set _options_ to ? GetOptionsObject(_options_).
         1. Let _opt_ be a new Record.
         1. Let _matcher_ be ? GetOption(_options_, *"localeMatcher"*, ~string~, &laquo; *"lookup"*, *"best fit"* &raquo;, *"best fit"*).
         1. Set _opt_.[[localeMatcher]] to _matcher_.
+        1. Let _sentenceBreakSuppressions_ be ? GetOption(_options_, *"sentenceBreakSuppressions"*, ~string~, &laquo; *"none"*, *"standard"* &raquo;, *"none"*).
+        1. Set _opt_.[[ss]] to _sentenceBreakSuppressions_.
         1. Let _localeData_ be %Segmenter%.[[LocaleData]].
         1. Let _r_ be ResolveLocale(%Segmenter%.[[AvailableLocales]], _requestedLocales_, _opt_, %Segmenter%.[[RelevantExtensionKeys]], _localeData_).
         1. Set _segmenter_.[[Locale]] to _r_.[[locale]].
+        1. Set _segmenter_.[[SentenceBreakSuppressions]] to _sentenceBreakSuppressions_.
```
Comment (sffc) [inline spec/segmenter.html:32]: I think you need to read from `_r_` here

```suggestion
        1. Set _segmenter_.[[SentenceBreakSuppressions]] to _r_.[[ss]].
```

**Review by ryzokuken — APPROVED** [review]
spec changes LGTM, none of the three mentioned extension keys are supported in `Intl` yet and we shouldn't introduce more than one in a single PR so it looks good to go from my end. Let's see how people feel about this.

**Comment by FrankYFTang** [comment]
Sorry, I think we need additional changes that I forget to mention in TG2. 

A. We should also change 
14.1.1 Intl.Locale ( tag [ , options ] )
https://tc39.es/ecma402/#sec-Intl.Locale
to take option
"sentenceBreakSuppressions" 

B. The internal slot of Locale should also include "ss"
https://tc39.es/ecma402/#sec-intl.locale-internal-slots

C. We should also add a getter in section 14.3 
https://tc39.es/ecma402/#sec-properties-of-intl-locale-prototype-object
14.3.X get Intl.Locale.prototype.sentenceBreakSuppressions

which return either "none" (default) or "standard")

**Comment by ben-allen** [comment]
Updated to incorporate required changes to Intl.Locale & include a note in Annex A. Unclear on needed changes to Annex B.

File: spec/locale.html [hunk 3]
```diff
@@ -52,6 +52,8 @@ <h1>Intl.Locale ( _tag_ [ , _options_ ] )</h1>
         1. If _numberingSystem_ is not *undefined*, then
           1. If _numberingSystem_ does not match the Unicode Locale Identifier `type` nonterminal, throw a *RangeError* exception.
         1. Set _opt_.[[nu]] to _numberingSystem_.
+        1. Let _sentenceBreakSuppressions_ be ? GetOption(_options_, *"sentenceBreakSuppressions"*, ~string~, &laquo;*"none"*, *"standard"* &raquo;, *"none"*).
```
Comment (FrankYFTang) [inline spec/locale.html:55]: For Intl.Locale constructor, the default value for "sentenceBreakSuppressions" should be undefined. (Comparing to other options, such as "calendar", "collation", "hourCycle", "caseFirst", "numeric", "numberingSystem")

**Comment by sffc** [comment]
Note: @dminor would like to wait until ICU4X supports this feature before merging this PR.

https://github.com/unicode-org/icu4x/issues/3927

**Pull request closed without merging**